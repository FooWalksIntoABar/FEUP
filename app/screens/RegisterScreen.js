import React, { Component } from 'react'
import { StatusBar, Text, View } from 'react-native'
import { Container, Content, Form, Item, Input, Label, Button } from 'native-base'

import styles from '../styles/ParentStyles/RegisterScreen.style'
import PropTypes from 'prop-types'
import EnvVars from '../constants/EnviromentVars'

export default class RegisterScreen extends Component {
  constructor (props) {
    super(props)
    this.state = {
      email: '',
      emailError: true,
      password: '',
      passwordError: true,
      emailHadInteraction: false,
      passwordHadInteraction: false
    }
  }

  handlePress (email, password) {
    if (!this.state.passwordError && !this.state.emailError) {
      fetch(EnvVars.apiUrl + 'routine_manager/register/', {
        method: 'POST',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json'
        },
        body: JSON.stringify({
          email: email,
          password: password
        })
      }).then((response) => response.json())
        .then((responseJson) => {
        if(responseJson.status === "200"){
            this.props.navigation.navigate('MainMenu')
        }
        else{
            //TODO: Falhou qql cena, não sei bem o que fazer aqui tbh
        }
          return responseJson
        })
        .catch((error) => {
          console.error(error)
        })
    }
  }

  validate (type, value) {
    if (type === 'email') {
      var re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
      if (re.test(value)) {
        this.setState(() => ({ emailError: false }))
      } else {
        this.setState(() => ({ emailError: true, email: value }))
      }
      this.setState(() => ({ emailHadInteraction: true, email: value }))
    } else if (type === 'password') {
      if (String(value).length < 6) {
        this.setState(() => ({ passwordError: true }))
      } else {
        this.setState(() => ({ passwordError: false }))
      }
      this.setState(() => ({ passwordHadInteraction: true, password: value }))
    }
  }

  render () {
    return (
      <Container style={styles.registerContainter}>
        <StatusBar hidden />
        <Content contentContainerStyle={styles.contentContainter}>
          <Text style={styles.registerTitle}>Registo de Conta</Text>
          <Form>
            <Item floatingLabel error={this.state.emailError && this.state.emailHadInteraction} success={!this.state.emailError && this.state.emailHadInteraction} style={styles.inputContainer}>
              <Label style={styles.labelText}>E-mail</Label>
              <Input
                className='email'
                onChangeText={(text) => this.validate('email', text.trim())}
                style={styles.labelText}
              />
            </Item>

            <Item floatingLabel error={this.state.passwordError && this.state.passwordHadInteraction} success={!this.state.passwordError && this.state.passwordHadInteraction} style={styles.inputContainer}>
              <Label style={styles.labelText}>Password</Label>
              <Input
                className='password'
                onChangeText={(text) => this.validate('password', text.trim())}
                style={styles.labelText}
                secureTextEntry
              />
            </Item>

            <Button rounded block primary style={styles.submitButton}>
              <Text style={styles.buttonText} onPress={() => this.handlePress(this.state.email, this.state.password)}>Concluir Registo</Text>
            </Button>
          </Form>

          <View style={styles.loginTextContainer}>
            <Text style={styles.loginText}>Já possui uma conta?</Text>
            <Text style={styles.loginRegisterText} onPress={() => this.props.navigation.navigate('RegisterMenu')}>Faça login aqui!</Text>
          </View>
        </Content>
      </Container>
    )
  }
}

RegisterScreen.propTypes = {
  navigation: PropTypes.object.isRequired
}
