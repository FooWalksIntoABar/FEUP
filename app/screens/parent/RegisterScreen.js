import React, { Component } from 'react'
import { StatusBar, Text, View } from 'react-native'
import { Container, Content, Form, Item, Input, Label, Button } from 'native-base'

import styles from '../../styles/ParentStyles/RegisterScreen.style'

export default class RegisterScreen extends Component {
  constructor (props) {
    super(props)
    this.state = {
      email: '',
      emailError: null,
      password: '',
      passwordError: null,
      emailHadInteraction: false,
      passwordHadInteraction: false
    }
  }

  validate (type, value) {
    if (type === 'email') {
      var re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
      if (re.test(value)) {
        this.setState(() => ({ emailError: false }))
      } else {
        this.setState(() => ({ emailError: true }))
      }
      this.setState(() => ({ emailHadInteraction: true }))
    } else if (type === 'password') {
      if (String(value).length < 6) {
        this.setState(() => ({ passwordError: true }))
      } else {
        this.setState(() => ({ passwordError: false }))
      }
      this.setState(() => ({ passwordHadInteraction: true }))
    }
  }

  render () {
    return (
      <Container style={styles.registerContainter}>
        <StatusBar hidden />
        <Content contentContainerStyle={styles.contentContainter}>
          <Text style={styles.registerTitle}>Registo de Conta</Text>
          <Form>
            <Item floatingLabel error={this.state.emailError} success={!this.state.emailError && this.state.emailHadInteraction} style={styles.inputContainer}>
              <Label style={styles.labelText}>E-mail</Label>
              <Input
                className='email'
                onChangeText={(text) => this.validate('email', text.trim())}
                style={styles.labelText}
              />
            </Item>

            <Item floatingLabel error={this.state.passwordError} success={!this.state.passwordError && this.state.passwordHadInteraction} style={styles.inputContainer}>
              <Label style={styles.labelText}>Password</Label>
              <Input
                className='password'
                onChangeText={(text) => this.validate('password', text.trim())}
                style={styles.labelText}
                secureTextEntry
              />
            </Item>

            <Button rounded block primary style={styles.submitButton}>
              <Text style={styles.buttonText}>Concluir Registo</Text>
            </Button>
          </Form>

          <View style={styles.loginTextContainer}>
            <Text style={styles.loginText}>Já possui uma conta?</Text>
            <Text style={styles.loginRegisterText}>Faça login aqui!</Text>
          </View>
        </Content>
      </Container>
    )
  }
}
