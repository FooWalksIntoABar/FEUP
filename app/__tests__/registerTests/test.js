import React from 'react'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import RegisterScreen from '../../screens/RegisterScreen'

configure({ adapter: new Adapter() })

describe('RegisterScreen snapshot', () => {
  it('renders RegisterScreen correctly', async () => {
    const wrapper = shallow(<RegisterScreen />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.find('.email').simulate('changeText', 'a')
    wrapper.find('.password').simulate('changeText', 'a')
    wrapper.instance().validate('email', 'email@aleatorio.pt')
    wrapper.instance().validate('password', 'Password$UP3Rsegura123456')
    wrapper.instance().validate('else', '')
    wrapper.unmount()
  })
})
