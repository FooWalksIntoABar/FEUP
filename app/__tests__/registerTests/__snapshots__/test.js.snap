// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterScreen snapshot renders RegisterScreen correctly 1`] = `
<Styled(Container)
  style={
    Object {
      "backgroundColor": "#3db3e2",
    }
  }
>
  <StatusBar
    animated={false}
    hidden={true}
    showHideTransition="fade"
  />
  <Styled(Content)
    contentContainerStyle={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "left": "10%",
        "width": "80%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "LinotteBold",
          "fontSize": 30,
          "marginBottom": "5%",
        }
      }
    >
      Registo de Conta
    </Text>
    <Styled(Form)>
      <Styled(Item)
        error={false}
        floatingLabel={true}
        style={
          Object {
            "marginBottom": "5%",
            "marginRight": "5%",
          }
        }
        success={false}
      >
        <Styled(Label)
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "LinotteBold",
              "fontSize": 20,
            }
          }
        >
          E-mail
        </Styled(Label)>
        <Styled(Input)
          className="email"
          onChangeText={[Function]}
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "LinotteBold",
              "fontSize": 20,
            }
          }
        />
      </Styled(Item)>
      <Styled(Item)
        error={false}
        floatingLabel={true}
        style={
          Object {
            "marginBottom": "5%",
            "marginRight": "5%",
          }
        }
        success={false}
      >
        <Styled(Label)
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "LinotteBold",
              "fontSize": 20,
            }
          }
        >
          Password
        </Styled(Label)>
        <Styled(Input)
          className="password"
          onChangeText={[Function]}
          secureTextEntry={true}
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "LinotteBold",
              "fontSize": 20,
            }
          }
        />
      </Styled(Item)>
      <Styled(Button)
        block={true}
        primary={true}
        rounded={true}
        style={
          Object {
            "marginTop": "10%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          onPress={[Function]}
          style={
            Object {
              "color": "white",
              "fontFamily": "LinotteBold",
              "fontSize": 20,
            }
          }
        >
          Concluir Registo
        </Text>
      </Styled(Button)>
    </Styled(Form)>
    <Component
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": "15%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "LinotteBold",
            "fontSize": 15,
            "marginBottom": "2%",
          }
        }
      >
        Já possui uma conta?
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "LinotteBold",
            "fontSize": 15,
            "marginBottom": "2%",
            "textDecorationLine": "underline",
          }
        }
      >
        Faça login aqui!
      </Text>
    </Component>
  </Styled(Content)>
</Styled(Container)>
`;
